libraries/rush-lib/src/api/RushProjectConfiguration.ts

```diff
@@ -299,12 +300,19 @@ export class RushProjectConfiguration {
    * Examines the list of source files for the project and the target phase and returns a reason
    * why the project cannot enable the build cache for that phase, or undefined if it is safe to so do.
    */
-  public getCacheDisabledReason(trackedFileNames: Iterable<string>, phaseName: string): string | undefined {
+  public getCacheDisabledReason(
+    trackedFileNames: Iterable<string>,
+    phaseName: string,
+    operation?: Operation
```

Comment:
Pass `isNoOp` directly? Or just have the caller check it? I'm not comfortable with a configuration file object (which RushProjectConfiguration is) taking a dependency on the execution logic layer, even only for types.

That'll also make your unit tests much simpler.